1.0/tools/tests.md

```diff
@@ -0,0 +1,385 @@
+(possibly unstable) element into your tests, you can implement a stub of the
+other element that always returns consistent data. 
+
+Use `replace()` to create stub elements. 
+
+    setup(function() {
+      replace('paper-button').with('fake-paper-button');
+    });
+
+For example, using the sample `replace()` above and the element below:
+
+    <dom-module id='x-el'>
+      <template>
+        <paper-button id="pb">button</paper-button>
+      </template>
+    </dom-module>
+
+At test runtime, the content template would be stamped out as:
+
+    <dom-module id='x-el'>
+      <template>
+        <fake-paper-button id="pb">button</fake-paper-button>
+      </template>
+    </dom-module>
+
+The attributes and content of the element are preserved, but the tag
+is replaced with the specified stub tag.
+
+Because the method is called within `setup()`, all of the changes are 
+reverted at the end of each test.
+
+## AJAX
+
+<div class="yt-embed">
+  <google-youtube
+    videoid="_9qARcdCAn4"
+    autoplay="0"
+    rel="0"
+    fluid>
+  </google-youtube>
+</div>
+
+`wct` includes [Sinon](http://sinonjs.org/), which enables you to mock XHR
+requests and create fake servers. 
+
+Below is an example of a simple XHR unit test suite for 
+[`<iron-ajax>`](https://elements.polymer-project.org/elements/iron-ajax).
+Check out Sinon's documentation for more in-depth examples. 
+
+{% highlight html %}
+<!-- create test fixture template -->
+<test-fixture id="simple-get">
+  <template>
+    <iron-ajax url="/responds_to_get_with_json"></iron-ajax>
+  </template>
+</test-fixture>
+<script>
+  suite('<iron-ajax>', function() {
+    var ajax;
+    var request;
+    var server;
+    var responseHeaders = {
+      json: { 'Content-Type': 'application/json' }
+    };
+    setup(function() {
+      server = sinon.fakeServer.create();
+      server.respondWith(
+        'GET',
+        /\/responds_to_get_with_json.*/, [
+          200,
+          responseHeaders.json,
+          '{"success":true}'
+        ]
+      );
+    });
+    teardown(function() {
+      server.restore();
+    });
+    suite('when making simple GET requests for JSON', function() {
+      setup(function() {
+        // get fresh instance of iron-ajax before every test
+        ajax = fixture('simple-get');
+      });
+      test('has sane defaults that love you', function() {
+        request = ajax.generateRequest();
+        server.respond();
+        expect(request.response).to.be.ok;
+        expect(request.response).to.be.an('object');
+        expect(request.response.success).to.be.equal(true);
+      });
+      test('has the correct xhr method', function() {
+        request = ajax.generateRequest();
+        expect(request.xhr.method).to.be.equal('GET');
+      });
+    });
+  });
+</script>
+{% endhighlight %}
+
+The example above uses Chai's [`expect`](http://chaijs.com/api/bdd/) assertion
+style. 
+{: .alert .alert-info }
+
+## Run a subset of tests {#test-subsets}
```

Comment:
I tend to use this feature to run all of my tests, not just a subset. @arthurevans wdyt?

Technically to run a subset you should be able to use Mocha's `only` option:

`describe.only(function() {...}`

but I think this may be broken in wct.